src/execution/operator/persistent/buffered_csv_reader.cpp

```diff
@@ -211,12 +194,20 @@ void BufferedCSVReader::SkipHeader(idx_t skip_rows, bool skip_header) {
 
 	if (skip_header) {
 		// ignore the first line as a header line
-		string read_line;
-		getline(*source, read_line);
-		linenr++;
+		InitParseChunk(sql_types.size());
+		ParseCSV(ParserMode::PARSING_HEADER);
 	}
 }
 
+// Helper function to generate column names
+static string GenerateColumnName(const idx_t total_cols, const idx_t col_number, const string prefix = "column") {
+	uint8_t max_digits = total_cols > 10 ? (int)log10((double)total_cols - 1) + 1 : 1;
```

Comment:
Rather than using log here I would use NumericHelper::UnsignedLength (from `src/include/duckdb/common/types/numeric_helper.hpp`). Log is a bit evil here and can potentially cause problems with floating point inaccuracies (unlikely with small values, though).

Reply:
Nice! Thanks